### ISSUE TITLE
Tile flexable

### ISSUE THREAD

**Issue by landaisbenj** [EdddieN/machinon-domoticz_theme#31]
OMG OMG OMGGGGGG !!!!!!!!!!!

I work hard and with help of aurora's code from blauwebuis, i find that:

`        if( $('div.divider').length > 1  ){
            $('div.divider:not(div[id="weatherwidgets"]:not(div[id="tempwidgets"])) > div.item').unwrap();
            $( ".container > div.item" ).wrapAll( "<div class='row divider' />");
        }
`
And now, i can do this: 

![38461462_268243263971469_3150691161128042496_n](https://user-images.githubusercontent.com/32144431/43898971-c8cdbcdc-9be0-11e8-8e1f-3e2d42f2fd91.gif)

Some bug may be fixed and stuff improve, but for now, i know how to do !!! 
It's the beggining. I'm so exited !!

**Comment by EdddieN** [comment]
Hey! you managed to put 4 tiles! Brilliant!! Great work 👍

**Comment by landaisbenj** [comment]
:D thanks but i need more time to make this real :( i try to pull the result on my vacation.

**Comment by landaisbenj** [comment]
only on css with this smart js code.

**Comment by EdddieN** [comment]
I think it could bu useful for the app mode when we reach that point

**Comment by DewGew** [comment]
Good work :)

**Comment by landaisbenj** [comment]
Yes but I seek about 1 week for detecting when dashcontent or lightcontent is loaded to apply my script. I'm not so far with mutation observer. I can detect all change but I don't know how target change on "main-view"

**Comment by DewGew** [comment]
Im curious about this.. how is it going??

**Comment by landaisbenj** [comment]
First i want to make the change in the light switch tab. So i must to detect when lightswitch finished it loaded. I can detect if we are on lightswitch page, i can detect the mutation on index.html but i don't know how detect when all tile are finished to load... 
If i want to make row divider change, i must know how row divider finish to load but i don't know how... Do you know how can we detect, after click on a tab (dashboard, lightcontent, weather...), the end of all js of the page ?

(btw that makes me question myself about click on setting tab. we can change the onclick tab to put another html code for setting page... This can pass through the custom page)

**Comment by landaisbenj** [comment]
Ok after many hour of reflexion i think i need to work with timeout :( 

But now i'm proud to show you my real first script in jquery (i think ?! xD) and this without known whatever in jquery on 2 mounth ago:

`	function DelRow(){
	$('#main-view div.row').each(function(){
		x=$(this).nextAll().children().detach();
		$(this).append(x).nextAll().remove();
	});
};

//detect le changement de page
function locationHashChanged() {
	setTimeout(function(){    
	if (location.hash === "#/LightSwitches" || $('#dashcontent')) {
       // console.log('test ok detection page change or #dashcontent appear');


       DelRow();
    }},500)

}

window.onhashchange = locationHashChanged;
`

this detect page change and suppress row for one. 
this can help me on my way to change container display to flex. 
Since to be continued..

**Comment by EdddieN** [comment]
That is great news! I must admit I admire your ability to code JS and similar... I never managed to get my head around it :D

**Comment by EdddieN** [comment]
Really looking forward to see it :)

**Comment by DewGew** [comment]
The french guy did a great job  #56 and #57 👍 :)

**Comment by EdddieN** [comment]
haha! Cannot wait to try it when I get home

**Comment by EdddieN** [comment]
You did it! 4 tiles in a row, well done!

**Comment by EdddieN** [comment]
is the attached to be expected or my cache messing up things? 3 columns with one big at the bottom
![screen shot 2018-09-03 at 20 14 13](https://user-images.githubusercontent.com/39434245/44999573-3b623280-afb6-11e8-8cde-0a7faa702cbc.png)

**Comment by landaisbenj** [comment]
Yes for the moment i don't I did not succeed to resolve this. But i think i can do it easy. Stay tune

**Comment by EdddieN** [comment]
I have been experimenting with a bit, I can see that jumps fro 4 to 3, to 2 then back to 3. Do you know how the decission of the size of the tiles is made?

![tiles](https://user-images.githubusercontent.com/39434245/45263843-67077180-b429-11e8-9d5b-8c69c6882bf5.gif)

**Comment by EdddieN** [comment]
See attached, is this switching from 3 to 2 and then to 3 normal? How does the script decide how many tiles should be at each moment?

![tiles](https://user-images.githubusercontent.com/39434245/45263865-e137f600-b429-11e8-8e66-d4f69f82cd3f.gif)

**Comment by landaisbenj** [comment]
It's à Bad way between media width and flex automatic width. I need tout clean custom.css

Benjamin

Envoyé de mon iPhone

Le 9 sept. 2018 à 13:14, machinon <notifications@github.com<mailto:notifications@github.com>> a écrit :


See attached, is this switching from 3 to 2 and then to 3 normal? How does the script decide how many tiles should be at each moment?

[tiles]<https://user-images.githubusercontent.com/39434245/45263865-e137f600-b429-11e8-8e66-d4f69f82cd3f.gif>

**Comment by Holland1** [comment]
To everyone that is working on this project, fantastic job, I really like the theme.

Just a cosmetic issue I would like to bring forward.

Dashboard and other tabs resize when the refresh button is pressed. This is browser independent.
A picture makes it clear. As can be seen below, a refresh results in a resize of the screen.

![switchs tabs](https://user-images.githubusercontent.com/13700844/45595514-7ba9b380-b9ad-11e8-86aa-ace9ec9a7595.JPG)

![refresh page](https://user-images.githubusercontent.com/13700844/45595516-849a8500-b9ad-11e8-8fbf-c11b92694e21.JPG)

**Comment by EdddieN** [comment]
This may help finding the issue:

1.- load from fresh (clear cache on switches), you will see 3 columns
2.- got to settings, this forces the flex layout
3.- go back to switches, now it loads 4

**Comment by landaisbenj** [comment]
Last merge on beta can fix the bug. Can you see that ?

**Comment by landaisbenj** [comment]
Option menu?

**Comment by EdddieN** [comment]
Menú -> Settings = no longer worked the machinon theme, also the Theme settings button no longer is there. Something broke on the latest beta merge that I reversed.

**Comment by DewGew** [comment]
I fixed the issue with latetest bete pull request. now it should work..

**Comment by DewGew** [comment]
@landaisbenj  Maybe you need to update the mobile section in custom.css to match your changes in the code below at row 1417:

> /* MOBILE */
> 
> table.mobileitem{
>     width: 100%;
>     padding: 0;
>     margin-top:20px;
>     border-bottom-right-radius: 0;
>     border-bottom-left-radius: 0;
> }
> table.mobileitem > thead{
>     display: flex;
> }
> table.mobileitem > thead > tr {
>     display:flex;
> 	background: none;
> }
> table.mobileitem > thead > tr > th:first-of-type{
>     flex-grow:1;
> }
> 
> table.mobileitem .status > span > span{margin-right:7px;}
> 
> .mobileitem th{
>     display:inline;
>     background-color: transparent;
>     text-shadow: none;
> }
> .mobileitem tr{
>     margin: 0!important;
>     padding: 0!important;
>     box-sizing: border-box;
>     height: auto;
>     display: flex;
>     flex-grow: 1;
>     flex-direction: row;
>     justify-content: space-between;
>     width: 100%;
> }
> .mobileitem td{
>     height: auto!important;
>     margin-right: 0!important;
>     /*word-break: break-word;*/
>     /*margin: 0!important;*/
>     background-image: none!important;
>     display: inline-flex;
>     width: auto;
>     flex-grow: 1;
>     flex-shrink: 1;
>     align-items: center; 
>     padding:5px!important;
>     min-height:35px;
> }
> .mobileitem td > div {margin: 0!important;}
> 
> .mobileitem td:last-child{
>     flex-wrap:wrap;
> }
> .mobileitem td label{
>     white-space: nowrap;
>     height:50%;
> }
> body.dashMobile table.mobileitem td#name,
> body.dashMobile table.mobileitem td.name{
>     font-size: 100%;
>     position: inherit;
>     padding:0 0 0 10px!important;
>     height:auto;
>     min-height:30px;
> }
> body.dashMobile table.mobileitem td#name img,
> body.dashMobile table.mobileitem td.name img{
>     margin-right:5px;
> }
> .mobileitem .btn-mini{
>     margin-left: 2px;
> }
> #dashcontent .mobileitem .dimslider{
>   background: rgba(0, 0, 0, 0.26);
>   border: none;
>   box-shadow: none;
>   height: 5px;
> }
> #dashcontent .mobileitem .dimslider .ui-slider-range{
>   background: rgba(11, 151, 205, 0.5);
> }
> #dashcontent .mobileitem .dimslider .ui-slider-handle{
>   width: 15px;
>   height: 15px;
>   background: var(--main-blue-color);
>   border-radius: 50%;
>   top: -6px;
> }
> #dashcontent .mobileitem td#status,
> #dashcontent .mobileitem td.status{
>     display: flex;
>     position:initial;
>     flex-direction: row;
>     white-space: normal;
>     width: auto;
>     max-width:60%;
>     justify-content: flex-end;
>     padding: 0;
>     flex-grow: 1;
>     flex-shrink: 1;
>     margin: 0;
>     text-align: end;
> }
> #dashcontent .mobileitem .selectorlevels {
>   position: relative;
>   right: 0px;
>   bottom: 0px;
> }

**Comment by DewGew** [comment]
This might be better .... Fix autoclose drop menu #127

**Comment by EdddieN** [comment]
....mmm have you tried this on a mobile device? On desktop all looks good (beta) but on mobile browser is all a bit out of place, like no menu
<img width="337" alt="screenshot 2018-10-05 at 10 06 38" src="https://user-images.githubusercontent.com/39434245/46526516-63c7a000-c886-11e8-9a2f-5caf7d259b43.png">

**Comment by landaisbenj** [comment]
Ok. Think loud.
 So needed:
Fix mozilla display,
Fix mobile observer
And mobile display

For this i need to more organize all custom.css. To do this, maybe i must do another branch or pause another subject to not interfere.

**Comment by DewGew** [comment]
@landaisbenj  Before you continue with this read this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/observe

It says: 

> Obsolete
This feature is obsolete. Although it may still work in some browsers, its use is discouraged since it could be removed at any time. Try to avoid using it.

**Comment by landaisbenj** [comment]
ok i have a first fix

**Comment by landaisbenj** [comment]
> @landaisbenj Before you continue, read this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/observe
> 
> It says:
> 
> > Obsolete
> > This feature is obsolete. Although it may still work in some browsers, its use is discouraged since it could be removed at any time. Try to avoid using it.

Just yet !!!! :'( Observer is new thing for me, so many time lost xD. Ok so i add one thing.

**Comment by landaisbenj** [comment]
I had two issue to follow change on observer object #129 and flex propriety  #130.

If you accept visual of last #128 close this.

**Comment by EdddieN** [comment]
Yes has been committed to beta so closing this then
